### PR TITLE
Add Third-Party RX Targets (Flywoo, JHEMCU, HiYounger)

### PR DESCRIPTION
**Pull request by coolchip**
I've using the Flywoo EL24E Receivers. The manufacturer mentions that you should use the DIY target in the configurator. 
https://flywoo.net/products/0-4g-flywoo-2-4g-expresslrs-elrs-el24e-el24p
This works for me. I just copied the DIY target and used it to create the target for the Flywoo receiver. I will also create a PR for the Configurator.

### CODE REVIEW

**Comment by deadbytefpv** [comment]
While you're at it, why not add the other's?
Like JHEMCU, HIYounger etc.. 

multiple birds with one PR..

**Comment by coolchip** [comment]
I do that very gladly. I will do that later in the evening.
JHEMCU, HIYounger... anything else? Will google for it later.

File: src/targets/flywoo_2400.ini [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+[env:Flywoo_EL24E_2400_RX_via_UART]
+extends = env:DIY_2400_RX_ESP8285_SX1280_via_UART
+
+[env:Flywoo_EL24E_2400_RX_via_BetaflightPassthrough]
+extends = env:DIY_2400_RX_ESP8285_SX1280_via_BetaflightPassthrough
+
+[env:Flywoo_EL24E_2400_RX_via_WIFI]
+extends = env:DIY_2400_RX_ESP8285_SX1280_via_WIFI
+
+[env:Flywoo_EL24P_2400_RX_via_UART]
```
Comment (JyeSmith) [inline src/targets/flywoo_2400.ini:15]: Do we need both E and P targets?  "Flywoo_EL24" should be enough.

**Comment by JyeSmith** [comment]
There is a lot of duplication here and manufacturers could have a single generic target.

**Comment by pkendall64** [comment]
I think I'd prefer is the base firmware had distinct targets based on capability and pin mapping, and configurator could have aliases for the hardware target and override the target short-name.

**Comment by coolchip** [comment]
This makes sense. Then I would reduce the targets as follows: 

Flywoo EL24E
Flywoo EL24P
-> Flywoo_EL24_2400_RX

JHEMCU SP24S
JHEMCU EP24S
-> JHEMCU_P24S_2400_RX

JHEMCU RX24T
-> JHEMCU_RX24T_2400_RX

HiYOUNGER SP24S
HiYOUNGER EP24S
-> HiYOUNGER_P24S_2400_RX

HiYOUNGER RX24T
-> HiYOUNGER_RX24T_2400_RX

In the configurator I would then enter all hardware modules of the manufacturers and refer to the corresponding generic targets.
Is that okay?

**Comment by pkendall64** [comment]
Not quite what I was meaning.
I would say we should have a
`BASE_ESP_TX_2400` which has feature flags for OLED, TFT, RGB etc
then in configurator we can specify the feature flags for the modules along with the DEVICE_NAME.
So all of the ones in this PR are either DIY or BETAFPV copies, so there should not need to be any changes in the main codebase other than an ifdef around the DEVICE_NAME in the header files.

**Comment by coolchip** [comment]
That's right. And that should significantly reduce the number of targets. But I don't know how to contribute to it. Sounds like a fundamental change.

**Comment by pkendall64** [comment]
I think we can leave it till 2.1 and I chat with the other devs about it.

**Comment by coolchip** [comment]
Okay. Please give a sign, when all devs agree. Then I will add the targets to the configurator.

**Comment by CapnBry** [comment]
Yeah I am against adding targets for every Tom, Dick, and Harry that clones the basic design and slaps their name on it. The configurator can have aliases but the PIO definitions should **not** have them. If you work in VS Code all day long working on different targets it is annoying to have to sort through 9,000 different target definitions to select the next one to test. Each PIO target added slows me down a little and can be frustrating when a new copy is added that just has a different name and it screws up my process because now the thing I am used to typing for getting to a "standard target" now gives me some other target.

In 2.1 we should be **removing** target definitions, not adding more for the exact same hardware and using aliases on the configurator side. This makes adding "targets" for these clones only reliant on changing one place: the configurator. This has the benefit of supporting "new" hardware (just old hardware with a new name) even on older versions.

**Comment by coolchip** [comment]
Okay. So I should add these Tom, Dick and Harry Boards to the configurator and let them point to the DIY/BetaFPV target? Or should I wait till 2.1?

**Comment by CapnBry** [comment]
You can do that right now since it isn't reliant on any code on the ELRS side. I don't think there is any target alias support in the configurator yet though so there would be actual work to be done there first.

**Comment by jurgelenas** [comment]
Will work on alias functionality asap :)

**Comment by pkendall64** [comment]
This is all handled in configurator, so closing now.

**Pull request closed without merging**